src/tools/interop/idt/README.md

```diff
@@ -0,0 +1,564 @@
+
+When the test completes, capture mode is stopped and all captured data is zipped
+in a file that can then be sent to all parties involved in investigating any
+issue uncovered via the manual test. Each ecosystem may implement an analysis
+that analyzes capture data, displays info to the user, probes the local
+environment and generates additional artifacts.
+
+## Getting started
+
+## Raspberry Pi installation
+
+### Environment overview
+
+The execution environment of IDT when using Raspberry Pi is shown in the figure
+below.
+
+[TODO] add figure.
+
+The Raspberry Pi is referred to as the "target environment" in this document. It
+is where all the "discovery" and "capture" work is performed.
+
+The "admin" computer is the machine used to connect to and control the RPi, and
+to fetch artifacts which were created during capture from the RPi.
+
+This directory contains tools for use on both the admin computer and the target
+environment.
+
+### Environment details
+
+1. This guide and the `idt` scripts expect **_one_** instance of `idt` on both
+   the admin computer and the target environment.
+1. The expected install location on the target environment is the home directory
+   of the user specified in `idt/scripts/vars.sh`. Default values are:
+    ```
+    PIHOST="kali-raspberry-pi"
+    PIUSER="kali"
+    ```
+1. Helper scripts may be used on admin computers that support `zsh` and `bash`
+   (Linux and macOS).
+1. Windows may be used as the admin computer via tools like `PowerShell`,
+   `MobaXterm` and `FileZilla`.
+1. This setup is intended to work with the admin computer and RPi connected to
+   the same Wi-Fi network, which is also the Wi-Fi network used for testing.
+1. Corporate networks are not expected to be used as test networks.
+
+### Prepare the RPi
+
+1. A >= 128 GB SD card is recommended.
+1. Flash the RPi SD with the latest
+   `Kali Linux RaspberryPi 2, 3, 4 and 400 (img.xz)` image (32 bit).
+    - Use [Raspberry Pi Imager](https://github.com/raspberrypi/rpi-imager) to
+      [download the image and prepare the SD](https://www.kali.org/docs/arm/using-rpi-imager-to-write-raspberry-pi-images/).
+1. Plug the SD into the RPi.
+1. Connect the RPi to your network via **ethernet.** [TODO] provide alternative
+1. Boot the RPi.
+
+### Configure admin computer and push to the RPi
+
+#### Linux and macOS admin computers
+
+1. On your admin computer, source the `alias` script from the parent directory
+   of `idt` to get `idt` commands in your current shell.
+    ```bash
+     source idt/scripts/alias.sh
+    ```
+    - To avoid having to repeat this step for each session, optionally configure
+      automatic aliases permanently.
+    - **_NOTE:_** Once run, `idt` commands will be globally and automatically
+      available. If you need to remove the installation, edit the `.rc` files
+      mentioned in `setup_shell`.
+    ```bash
+    source idt/scripts/setup_shell.sh
+    ```
+1. Send `idt` to the RPi:
+    ```
+    idt_push
```

Comment:
It looks like there are some assumptions made by the script that may not be valid. It looks like you assume that venv was setup. I think there are venv instructions further down, but I think we should update the script without the built-in assumption. I think we shouldn't make the tool depend on a specific Python installation style.

```bash
ubuntu@ubuntu:~/connectedhomeip/src/tools/interop$ idt_push
mv: cannot stat 'idt/venv': No such file or directory
kali@192.168.1.50's password:
scp: /home/kali/idt: No such file or directory
mv: cannot stat 'TEMPvenv': No such file or directory
```

Reply:
Yes this was just a quick provision to ensure I didn't copy my dev machine venv to the RPi for no reason.
As the script is sourced, the error has no outcome if venv isn't there, but I've made it better as suggested anyway! 
I noted your advice about `mktemp` (thanks!) but think I missed the use case in this specific context.